restdocs-api-spec-openapi3-generator/src/test/kotlin/com/epages/restdocs/apispec/openapi3/OpenApi3GeneratorTest.kt

```diff
@@ -178,14 +178,47 @@ class OpenApi3GeneratorTest {
     }
 
     @Test
-    fun `should extract multiple parameters when seperated by delimiter`() {
+    fun `should extract multiple parameters when separated by delimiter`() {
         givenResourceWithMultiplePathParameters()
 
         whenOpenApiObjectGenerated()
 
         thenMultiplePathParametersExist()
     }
 
+    @Test
+    fun `should handle urlencoded body in POST request as a request body not query string`() {
+        val method = HTTPMethod.POST
+
+        givenResourceWithFormDataSentAs(method)
+
+        whenOpenApiObjectGenerated()
+
+        thenResourceHasFormDataInRequestBodyNotAsQueryParameters(method.toString().toLowerCase())
+    }
+
+    @Test
+    fun `should handle urlencoded body in PUT request as a request body not query string`() {
+        val method = HTTPMethod.PUT
+
+        givenResourceWithFormDataSentAs(method)
+
+        whenOpenApiObjectGenerated()
+
+        thenResourceHasFormDataInRequestBodyNotAsQueryParameters(method.toString().toLowerCase())
+    }
+
+    fun thenResourceHasFormDataInRequestBodyNotAsQueryParameters(method: String) {
+        val productGetByIdPath = "paths./products/{id}-{subId}.$method"
```

Comment:
This would need to be adapted to `/products/{id}` if the suggestion below is followed